### PR TITLE
Zoom reports

### PR DESCRIPTION
**Pull request by averbn**
Fixes #

## Proposed Changes

  - Adding ZoomReports data connector

### CODE REVIEW

**Review by v-maudan — CHANGES_REQUESTED** [review]

File: DataConnectors/ZoomReports/azuredeploy_Connector_ZoomAPI_AzureFunction.json [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                        "APPLICATIONINSIGHTS_CONNECTION_STRING": "[reference(resourceId('microsoft.insights/components', variables('FunctionName')), '2015-05-01').ConnectionString]",
+                        "AzureWebJobsStorage": "[concat('DefaultEndpointsProtocol=https;AccountName=', toLower(variables('FunctionName')),';AccountKey=',listKeys(resourceId('Microsoft.Storage/storageAccounts', toLower(variables('FunctionName'))), '2019-06-01').keys[0].value, ';EndpointSuffix=',toLower(variables('StorageSuffix')))]",
+                        "logAnalyticsUri": "[variables('LogAnaltyicsUri')]",
+                        "WorkspaceID": "[parameters('WorkspaceID')]",
+                        "WorkspaceKey": "[parameters('WorkspaceKey')]",
+                        "ZoomApiKey": "[parameters('ZoomApiKey')]",
+                        "ZoomApiSecret": "[parameters('ZoomApiSecret')]",
+                        "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/averbn/azure_sentinel_data_connectors/blob/main/zoom-reports-azure-sentinel-data-connector/ZoomAPISentinelConn.zip?raw=true"
```
[the remark this answers is not in the document]
Comment (averbn) [reply DataConnectors/ZoomReports/azuredeploy_Connector_ZoomAPI_AzureFunction.json:157]: done

File: DataConnectors/ZoomReports/ZoomReports_API_FunctionApp.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	],
+	"connectivityCriterias": [{
+			"type": "IsConnectedQuery",
+			"value": [
+				"Zoom_CL\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
+			]
+		}
+	],
+	"availability": {
+		"status": 1,
```
[the remark this answers is not in the document]
Comment (averbn) [reply DataConnectors/ZoomReports/ZoomReports_API_FunctionApp.json:31]: done

File: DataConnectors/ZoomReports/ZoomReports_API_FunctionApp.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+						],
+						"label": "Primary Key"
+					},
+					"type": "CopyableLabel"
+				}
+			]
+		},
+		{
+			"title": "Option 1 - Azure Resource Manager (ARM) Template",
+			"description": "Use this method for automated deployment of the Zoom Audit data connector using an ARM Tempate.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/sentinel-ZoomAPI-azuredeploy)](https://aka.ms/sentinel-ZoomAPI-azuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n> **NOTE:** Within the same resource group, you can't mix Windows and Linux apps in the same region. Select existing resource group without Windows apps in it or create new resource group.\n3. Enter the **ZoomApiKey**, **ZoomApiSecret** and deploy. \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy."
```
[the remark this answers is not in the document]
Comment (averbn) [reply DataConnectors/ZoomReports/ZoomReports_API_FunctionApp.json:106]: done

File: Parsers/ZoomReports/Zoom.txt [hunk 4]
```diff
@@ -0,0 +1,107 @@
+// Usage Instruction : 
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name and alias (e.g. Zoom).
+// Function usually takes 10-15 minutes to activate. You can then use function alias from any other queries (e.g. Zoom | take 10).
+// References : 
+// Using functions in Azure monitor log queries : https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+// Tech Community Blog on KQL Functions : https://techcommunity.microsoft.com/t5/Azure-Sentinel/Using-KQL-functions-to-speed-up-analysis-in-Azure-Sentinel/ba-p/712381
+//
+let Zoom_dates_report_view = view () {
+    Zoom_CL 
+    | where event_type_s == "dates"
```
Comment (v-maudan) [inline Parsers/ZoomReports/Zoom.txt:10]: Usually good to use "=~" to allow for the string value match to not be case sensitive

**Review by v-maudan — DISMISSED** [review]

File: Parsers/ZoomReports/Zoom.txt (hunk 4 above)
Comment (averbn) [reply Parsers/ZoomReports/Zoom.txt:10]: Thanks, but in this case, it is not necessary because data types exactly predefined in the python code.

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**